ch_pipeline/core/dataquery.py

```diff
@@ -711,25 +713,43 @@ def next(
 
             if len(unique_dataset_ids) != 0:
                 fmaps = {}
-                # Get the frequency map for each unique dataset_id. They should all
+                # Get the frequency map for each unique dataset id. They should all
                 # have the same frequency map - otherwise, raise an error
                 for dsid in unique_dataset_ids:
-                    state = (
-                        ds.Dataset.from_id(str(dsid))
-                        .closest_ancestor_of_type("f_engine_frequency_map")
-                        .state
+                    _missing_ds_flag = False
+                    try:
+                        state = (
+                            ds.Dataset.from_id(str(dsid))
+                            .closest_ancestor_of_type("f_engine_frequency_map")
+                            .state
+                        )
+                        fmaps[state.id] = state.data
+                    except exceptions.NotFoundError:
+                        # No frequency map found for this dataset id, so the default
+                        # is used. Provide an entry so we can still check how many
+                        # frequency maps were found
+                        fmaps["default"] = 1
+                    except DoesNotExist:
+                        # This dataset id does not exist in the database. If none of the
+                        # dataset ids are found, throw an exception
+                        _missing_ds_flag = True
+                        continue
```

Comment:
Peewee could throw a `DoesNotExist` at many stages. I think being a little more surgical with the exception checking might help here. Something like:

```suggestion
                    _missing_ds_flag = False
                    try:
                        dataset = ds.Dataset.from_id(str(dsid))
                    except DoesNotExist:
                        # This dataset id does not exist in the database. If none of the
                        # dataset ids are found, throw an exception
                        _missing_ds_flag = True
                        continue
                        
                    try:
                        state = dataset.closest_ancestor_of_type("f_engine_frequency_map").state
                        fmaps[state.id] = state.data
                    except exceptions.NotFoundError:
                        # No frequency map found for this dataset id, so the default
                        # is used. Provide an entry so we can still check how many
                        # frequency maps were found
                        fmaps["default"] = 1
```

Reply:
Good idea